src/Razor/src/Microsoft.AspNetCore.Razor.Language/Components/ComponentGenericTypePass.cs

```diff
@@ -275,7 +275,7 @@ private void CreateTypeInferenceMethod(DocumentIntermediateNode documentNode, Co
 
                     // Method name is generated and guaranteed not to collide, since it's unique for each
                     // component call site.
-                    MethodName = $"Create{node.TagName}_{_id++}",
+                    MethodName = $"Create{CSharpIdentifier.SanitizeIdentifier(node.TagName)}_{_id++}",
```

Comment:
Can't use the tag name directly anymore because it can be a fully qualified tag name which contains `.`